src/steps/fetchTeams.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  createIntegrationEntity,
+} from '@jupiterone/integration-sdk';
+import { requestAll } from '../pager-duty';
+import { Team } from '../types';
+
+const step: IntegrationStep = {
+  id: 'fetch-teams',
+  name: 'Fetch Teams',
+  types: ['pagerduty_team'],
+  async executionHandler({
+    logger,
+    jobState,
+    instance,
+  }: IntegrationStepExecutionContext) {
+    const { apiKey } = instance.config;
+
+    logger.info('Requesting /teams endpoint');
+
+    const teams = await requestAll<Team>('/teams', 'teams', apiKey);
```

Comment:
There is no need to change this (or other similar such uses of `requestAll`) now, but we've been finding that an `iterateAll` approach, where that function would paginate and callback - just as `iterateEntities` does - for each object, before calling for the next page, and where the callback processes the provider object completely, allows for scaling up in customer accounts which may have a significant number of objects. This pattern is essentially a kind of stream processing model, where we shuffle data from the provider into local storage in a page-at-a-time fashion that minimizes memory consumption, and also allows us to collect some data even should later pages lead to a failure of some sort.

Reply:
This is a great idea I'm going to make a ticket to make this change, just to get this initial implementation done.